dashboard/app/controllers/lessons_controller.rb

```diff
@@ -41,14 +43,21 @@ def update
   private
 
   def lesson_params
+    # Convert camelCase params to snake_case. Right now this only works on
+    # top-level key names. This lets us do the transformation before calling
+    # .permit, so that we can use snake_case key names in our parameter list,
+    # because transform_keys returns a params object while deep_transform_keys
```

Comment:
Do intend to keep nested keys as-is without conversion? I think the comment should just say what the behavior should be, for both top level and nested keys, and we can leave it up to future edits to make sure we're implementing it properly.